### PR TITLE
CSD-81 Mission statement paragraph type with basic pattern and styles

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Mission statement paragraph type and stubbed out pattern.

# Need Review By (Date)
- 5/22

# Urgency
- medium

# Steps to Test
1. `composer require SU-HKKU/cardinal_service_profile:dev-CSD-81`
1. `blt drupal:sync --site=cardinalservice`
1. edit or create a basic page
1. add the "mission statement" tool and populate the fields
1. verify it works. styling is done in another task.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: themes/cardinal_service/lib/components/mission-statement/mission_statement.scss [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    color: #fff;
+    width: 100%;
+    height: 100%;
+    text-align: center;
+  }
+
+  .overlay-text {
+    text-align: center;
+    margin: 0 auto;
+    max-width: 1426px;
```
Comment (joegl) [inline themes/cardinal_service/lib/components/mission-statement/mission_statement.scss:33]: I'm curious about the max-width of 1426px

File: themes/cardinal_service/lib/components/mission-statement/mission_statement.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  label: Mission Statement
+  description: 'Display text and a link over a background image.'
+  fields:
+    image:
+      type: image
+      label: Image
+      description: 'Title of the opportunity.'
+      preview:
+        theme: image
+        uri: http://placecorgi.com/800x1200
```
Comment (joegl) [inline themes/cardinal_service/lib/components/mission-statement/mission_statement.ui_patterns.yml:11]: Never seen this before haha

File: tests/codeception/acceptance/Paragraphs/MissionStatementCest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          'admin_title' => 'Card',
+        ]),
+      ],
+    ]);
+
+    $I->amOnPage($node->toUrl()->toString());
+    $I->canSee('This is the mission statement whether you choose to accept it or not');
+    $I->canSeeLink('Verify your identity');
+    $I->canSee('This paragraph will self destruct at the end of this test');
+  }
```
Comment (joegl) [inline tests/codeception/acceptance/Paragraphs/MissionStatementCest.php:42]: Should there be a test for the C2A URI?

File: themes/cardinal_service/lib/components/mission-statement/mission_statement.scss (hunk 1 above)
Comment (pookmish) [reply themes/cardinal_service/lib/components/mission-statement/mission_statement.scss:33]: Its just a placeholder for now. Rebecca will be styling it and make it look all better.

File: themes/cardinal_service/lib/components/mission-statement/mission_statement.ui_patterns.yml (hunk 2 above)
Comment (pookmish) [reply themes/cardinal_service/lib/components/mission-statement/mission_statement.ui_patterns.yml:11]: its my favorite :smile:

File: tests/codeception/acceptance/Paragraphs/MissionStatementCest.php (hunk 3 above)
Comment (pookmish) [reply tests/codeception/acceptance/Paragraphs/MissionStatementCest.php:42]: sounds good to me :+1:

Comment (pookmish) [reply tests/codeception/acceptance/Paragraphs/MissionStatementCest.php:44]: https://github.com/SU-HKKU/cardinal_service_profile/pull/23/commits/5fb1c99bf658a441cfc913bd3ec59f74a91e9a2b#diff-649603cd48bdfc49ea5d40f7b4e2482bR42

**Pull request merged**